testutil/keeper/proof.go

```diff
@@ -45,50 +58,129 @@ func ProofKeeper(
 
 	ctrl := gomock.NewController(t)
 	mockSessionKeeper := mocks.NewMockSessionKeeper(ctrl)
-	mockSessionKeeper.EXPECT().
-		GetSession(gomock.Any(), gomock.Any()).
-		DoAndReturn(
-			func(
-				ctx context.Context,
-				req *sessiontypes.QueryGetSessionRequest,
-			) (*sessiontypes.QueryGetSessionResponse, error) {
-				session, ok := sessionByAppAddr[req.GetApplicationAddress()]
-				require.Truef(t, ok, "application address not provided during mock construction: %q", req.ApplicationAddress)
-
-				return &sessiontypes.QueryGetSessionResponse{
-					Session: &sessiontypes.Session{
-						Header: &sessiontypes.SessionHeader{
-							ApplicationAddress:      session.GetApplication().GetAddress(),
-							Service:                 req.GetService(),
-							SessionStartBlockHeight: 1,
-							SessionId:               session.GetSessionId(),
-							SessionEndBlockHeight:   5,
-						},
-						SessionId:           session.GetSessionId(),
-						SessionNumber:       1,
-						NumBlocksPerSession: session.GetNumBlocksPerSession(),
-						Application:         session.GetApplication(),
-						Suppliers:           session.GetSuppliers(),
-					},
-				}, nil
-			},
-		).AnyTimes()
+	mockAppKeeper := mocks.NewMockApplicationKeeper(ctrl)
+	mockAccountKeeper := mocks.NewMockAccountKeeper(ctrl)
 
 	k, err := keeper.NewKeeper(
 		cdc,
 		runtime.NewKVStoreService(storeKey),
 		log.NewNopLogger(),
 		authority.String(),
 		mockSessionKeeper,
-		nil,
-		nil,
+		mockAppKeeper,
+		mockAccountKeeper,
 	)
 	require.NoError(t, err)
 
 	ctx := sdk.NewContext(stateStore, cmtproto.Header{}, false, log.NewNopLogger())
 
-	// Initialize params
-	require.NoError(t, k.SetParams(ctx, types.DefaultParams()))
-
 	return k, ctx
 }
+
+func NewProofKeeperWithDeps(t testing.TB) (ProofKeeperWithDeps, context.Context) {
+	t.Helper()
+
+	// Collect store keys for all keepers which be constructed & interact with the state store.
+	keys := storetypes.NewKVStoreKeys(
+		types.StoreKey,
+		sessiontypes.StoreKey,
+		suppliertypes.StoreKey,
+		apptypes.StoreKey,
+		gatewaytypes.StoreKey,
+	)
+
+	// Construct a multistore & mount store keys for each keeper that will interact with the state store.
+	stateStore := integration.CreateMultiStore(keys, log.NewNopLogger())
+
+	logger := log.NewTestLogger(t)
+	ctx := sdk.NewContext(stateStore, cmtproto.Header{}, false, logger)
+
+	// Set block height to 1 so there is a valid session on-chain.
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	ctx = sdkCtx.WithBlockHeight(1)
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
+	authority := authtypes.NewModuleAddress(govtypes.ModuleName)
+
+	// Mock the bank keeper.
+	ctrl := gomock.NewController(t)
```

Comment:
Why did we decide to mock the bank keeper rather than using a real one?

Reply:
The bank keeper isn't used by the proof keeper or its dependencies. We may be able to remove it as a dependency; however, it seems like the bank keeper is a dependency by default as a result generating modules with ignite. Further investigation would be necessary.